hail/python/hail/backend/backend.py

```diff
@@ -76,6 +76,10 @@ def parse_vcf_metadata(self, path):
     def fs(self):
         pass
 
+    @abc.abstractmethod
+    def index_bgen(self, files, index_file_map, rg, contig_recoding, skip_invalid_loci):
```

Comment:
can this be an IR node instead?

Reply:
Good question.  Eventually, I think it has to: it calls Spark and it has to be lowered down to CDA and the infrastructure that targets batch.